### ISSUE TITLE
Website is too complex (eco-design)

### ISSUE THREAD

**Issue by sduperret** [awesome-selfhosted/awesome-selfhosted-data#995]
Hello everyone,

I am creating this issue to start a reflection on the complexity of displaying the website index page, which contains more than 33,000 items according to this report: https://www.ecoindex.fr/en/result/?id=145bce69-110c-4e6a-a68f-0a800352e69b

I think it would be beneficial for those of us who have a low-powered machine, or a bad internet connection, to try lazy loading or limiting the display to a home page and not the full list.

What do you think?

Link of the report (not permanent) : https://www.ecoindex.fr/en/result/?id=145bce69-110c-4e6a-a68f-0a800352e69b
To make a new report : https://www.ecoindex.fr/ (and then type the url of the website)

**Comment by sduperret** [comment]
poke @untypequicode

**Comment by nodiscc** [comment]
Hi,

> to try lazy loading

Our current static site rendering tool is [sphinx](https://www.sphinx-doc.org/en/master/) which doesn't support "lazy-loading" the DOM.

Adding this feature would be a complex task, and would probably require the addition of javascript to a relatively JS-free site (it even works with JS disabled), which would come at the cost of additional CPU cycles on all clients, costs which are, in my humble and baseless opinion, greater than those of "just" loading a large DOM.

That said, if someone would like to go through the process of adding this feature to sphinx, and provide a decent comparative benchmark with loading times/bandwidth use/CO2eq footprint, I would consider adding it. IMHO just the food, coffee and electricity consumed working on this task would negate any possible energy savings in the grand scheme of things; but I'd like to be proven wrong.

Strictly speaking about the bandwidth, the HTML weighs 230kb, followed by the favicon at 17kb, followed by the furo and sphinx-design stylehseets at 17 and 10kb. But I realize this discussion is about DOM complexity, which possibly could (how?) be optimized.

The 3 svg badges are only 10-15kb but trigger supplementary HTTP requests to different hosts and browser SVG rendering code, which probably (again, how to get a sensible measurement?) amount to bigger energy costs and loading times than DOM parsing.

In fact, if you run the Firefox performance profiler, you quickly realize that most CPU time is *not* spent on DOM  rendering but on other tasks.

![image](https://github.com/user-attachments/assets/0a8669c0-de29-4d9e-9bd6-b0254ff49e7c)

In short, I think purely looking at DOM complexity is the wrong approach, even if I agree with the premise of the tool (simple websites are better).

> low-powered machine, or a bad internet connection

Is there any evidence that the site is slow to load on e.g. a low-powered ARM box? It is quite fast even on my old-ass Samsung phone from 2016 which struggles on the majority of websites. Again, what potential load time/bandwidth/energy savings are we talking about?

> display to a home page and not the full list

What do you suggest? A relatively empty page with a single "Click here to see the full list" would defeat the whole purpose of this issue.

**Comment by n3storm** [comment]
I think an sphinx theme with some modifications to include HTMLX (https://htmx.org/) could help achieve this goal.

BTW I agree site is not slow, or overload even on cellphone connections. On the other hand using navigation is not intutive (but I guess that's another issue)

**Comment by guest20** [comment]
More JavaScript to make the site less complex?

**Comment by nodiscc** [comment]
No reply for several months, and no further explanation of what "too complex" means in this context.
Closing.
